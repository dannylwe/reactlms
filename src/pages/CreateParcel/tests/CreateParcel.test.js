import React from "react";
import renderer from "react-test-renderer";
import { mount, shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import Parcel, { CreateParcel } from "../CreateParcel";
import { Provider } from "react-redux";
import store from "../../../../store";

describe("CreateParcel", () => {
  it("should render without failing", () => {
    const wrapper = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <Parcel />
        </MemoryRouter>
      </Provider>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should handle changes", () => {
    const wrapper = shallow(<CreateParcel />);
    const spy = jest.spyOn(wrapper.instance(), "handleChange");
    wrapper
      .find("#pickupInput")
      .simulate("change", { target: { pickup: "test_pickup" } });
    wrapper
      .find("#destinationInput")
      .simulate("change", { target: { destination: "test_destination" } });
    wrapper
      .find("#nicknameInput")
      .simulate("change", { target: { nickname: "test_nickname" } });
    wrapper
      .find("#weightInput")
      .simulate("change", { target: { weight: "5" } });
    wrapper
      .find("#heightInput")
      .simulate("change", { target: { height: "6" } });
    expect(spy).toHaveBeenCalled();
  });
  it("should call submit", () => {
    const props = {
      createParcelAction: jest.fn()
    };
    const wrapper = shallow(<CreateParcel {...props} />);
    const spy = jest.spyOn(wrapper.instance(), "onSubmit");
    const event = { preventDefault: jest.fn() };
    wrapper.instance().onSubmit(event);
    expect(spy).toHaveBeenCalled();
  });
});
