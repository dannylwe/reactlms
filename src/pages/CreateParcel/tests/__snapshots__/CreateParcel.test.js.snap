// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateParcel should render without failing 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/dashboard"
    >
      Home
    </a>
    <a
      className="navbar-brand"
      href="/create"
    >
      Create Parcel
    </a>
    <a
      className="navbar-brand"
      href="/login"
      onClick={[Function]}
    >
      Logout
    </a>
  </nav>,
  <div
    className="create-title"
  >
    <h4>
      <p>
        Create Parcel
      </p>
    </h4>
  </div>,
  <div
    className="form-create"
  >
    <form
      onSubmit={[Function]}
    >
      <div>
        <label>
          Pickup location
        </label>
        <input
          aria-describedby="pickupHelp"
          className="form-control"
          id="pickupInput"
          name="pickup"
          onChange={[Function]}
          placeholder="Enter Pickup Location"
          type="text"
        />
      </div>
      <div>
        <label>
          Destination
        </label>
        <input
          aria-describedby="destinationHelp"
          className="form-control"
          id="destinationInput"
          name="destination"
          onChange={[Function]}
          placeholder="Destination"
          type="text"
        />
      </div>
      <div>
        <label>
          Nickname
        </label>
        <input
          aria-describedby="nicknameHelp"
          className="form-control"
          id="nicknameInput"
          name="nickname"
          onChange={[Function]}
          placeholder="Pichachu"
          type="text"
        />
      </div>
      <div>
        <label>
          Weight
        </label>
        <input
          aria-describedby="weightHelp"
          className="form-control"
          id="weightInput"
          name="weight"
          onChange={[Function]}
          placeholder="weight in kg < 10"
          type="number"
        />
      </div>
      <div>
        <label>
          Height
        </label>
        <input
          aria-describedby="heightHelp"
          className="form-control"
          id="heightInput"
          name="height"
          onChange={[Function]}
          placeholder="Height in kg < 10"
          type="number"
        />
      </div>
      <div
        className="submit-button"
      >
        <button
          className="btn btn-primary submit-btn"
          id="parcelButton"
          type="submit"
        >
          Create Parcel
        </button>
      </div>
    </form>
  </div>,
]
`;
